### PR TITLE
Added TestPanicsWithValue

### PR DESCRIPTION
**Pull request by nicolopignatelli**

### CODE REVIEW

File: assert/assertions.go [hunk 1]
```diff
@@ -606,6 +606,22 @@ func Panics(t TestingT, f PanicTestFunc, msgAndArgs ...interface{}) bool {
...
+// PanicsWithValue asserts that the code inside the specified PanicTestFunc panics with a specified value.
+//
+//   assert.Panics(t, func(){
+//     GoCrazy()
+//   }, "Panic value", "Calling GoCrazy() should panic with value \"Panic value\"")
+//
+// Returns whether the assertion was successful (true) or not (false).
+func PanicsWithValue(t TestingT, f PanicTestFunc, v interface{}, msgAndArgs ...interface{}) bool {
+
+	if funcDidPanic, panicValue := didPanic(f); !funcDidPanic || panicValue != v {
```
Comment (ernesto-jimenez) [inline assert/assertions.go]: Should use `Equal` to assert the value since we are comparing two interfaces.

File: assert/assertions.go [hunk 2]
```diff
@@ -606,6 +606,22 @@ func Panics(t TestingT, f PanicTestFunc, msgAndArgs ...interface{}) bool {
...
 }
 
+// PanicsWithValue asserts that the code inside the specified PanicTestFunc panics with a specified value.
+//
+//   assert.Panics(t, func(){
+//     GoCrazy()
+//   }, "Panic value", "Calling GoCrazy() should panic with value \"Panic value\"")
+//
+// Returns whether the assertion was successful (true) or not (false).
+func PanicsWithValue(t TestingT, f PanicTestFunc, v interface{}, msgAndArgs ...interface{}) bool {
```
Comment (ernesto-jimenez) [inline assert/assertions.go]: Most method's signatures have `expected, actual`. I'm thinking that the order should be `PanicsWithValue(t, "value", func() {})` for consistency.

**Comment by ernesto-jimenez** [comment]
@nicolopignatelli beyond the two comments in the source, right now it's also necessary to define a `PanicsWithValue` function in three more places with it's corresponding tests.

e.g:
```
./assert/assertions.go:func Equal(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) bool {
./assert/forward_assertions.go:func (a *Assertions) Equal(expected, actual interface{}, msgAndArgs ...interface{}) bool {
./require/forward_requirements.go:func (a *Assertions) Equal(expected, actual interface{}, msgAndArgs ...interface{}) {
./require/requirements.go:func Equal(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) {
```

Thanks!
Ernesto

**Comment by theory** [comment]
Oh man I could use this test! I have a lot of places to test panics, and I want to make sure things panic the way the tests think they should.

**Comment by nicolopignarocket** [comment]
@ernesto-jimenez please check again and tell me if we are going to switch parameters order or leave it like this

**Comment by nicolopignarocket** [comment]
ps: I used my work account for this last commit, I will clean things up when we are ready to squash

**Comment by nicolopignatelli** [comment]
@ernesto-jimenez <https://github.com/ernesto-jimenez> please check again
and tell me if we are going to switch parameters order or leave it like
it's now

Il giorno mer 2 dic 2015 alle ore 16:47 Nicolo Pignatelli <
notifications@github.com> ha scritto:

**Comment by boyan-soubachov** [comment]
I am closing this PR due to its age. If you wish to reopen and rebase, I'd be happy to review.

**Pull request closed without merging**